### ISSUE TITLE
volume change on reboot, distortion, wildly jumping volume with compressor preset

### ISSUE THREAD

**Issue by atomGit** [wwmm/easyeffects#1994]
### EasyEffects Version

7.0.0

### What package are you using?

Arch (easyeffects)

### Distribution

Manjaro

### Describe the bug

* output volume not rememberd on reboot
* distortion and wildly changing volume when using [EasyEffects-Presets/LoudnessEqualizerPE.json at master · Digitalone1/EasyEffects-Presets](https://github.com/Digitalone1/EasyEffects-Presets/blob/master/LoudnessEqualizer.json)

these are separate issues

when easy is enabled, the volume may be different after a reboot (louder for me) - adjusting the volume, even a tiny amount, causes the level to jump back to normal and all seems ok again

when using the compressor preset the problem is amplified many times - the volume can jump *wildly* when the volume slider is manually adjusted and there is severe distortion in the audio - this can be traced to the gate component; if the gate is disabled, all returns to normal

the common thread in these issues is the volume changing randomly 

Manjaro KDE/Plasma
kernel 6.0.8
easyeffects 7.0.0 (arch repo)
pipewire 1:0.3.59 (arch repo)
alsa-ucm-conf  1.2.7.1  (arch repo - i can't upgrade due to [this](https://github.com/alsa-project/alsa-ucm-conf/issues/236) bug)

### Expected Behavior

* volume should be remembered across reboot
* volume should not jump randomly when adjusted and compressor preset is in play
* distortion should not occur when gate component of compressor preset is in play

### Debug Log

<details><summary>Debug Log</summary><br/>

```
Paste your log here
```
</details>

### Additional Information

[audio distortion due to gate and system upgrade · Issue #11 · Digitalone1/EasyEffects-Presets](https://github.com/Digitalone1/EasyEffects-Presets/issues/11)

**Comment by wwmm** [comment]
>volume should be remembered across reboot

EasyEffects isn't the one that should remember system volumes. This is the audio server (PipeWire) responsibility. 

>volume should not jump randomly when adjusted and compressor preset is in play
>distortion should not occur when gate component of compressor preset is in play

These two plugins are essentially the same they were many months ago. Either this preset has something that is not playing nice with the latest EasyEffects release or something odd happened in your PipeWire install.

**Comment by wwmm** [comment]
I tried this preset here
```
"output": {
        "plugins_order": [
            "gate",
            "compressor",
            "multiband_compressor",
            "equalizer",
            "limiter",
            "bass_enhancer",
            "exciter",
            "autogain",
            "multiband_gate",
            "convolver",
            "crystalizer",
            "stereo_tools",
            "reverb",
            "delay",
            "deesser",
            "crossfeed",
            "loudness",
            "maximizer",
            "filter",
            "pitch"
        ]
```
Although this preset loads I think it needs to be updated to our current release. It is strange that all these plugins are in the list.

**Comment by atomGit** [comment]
> These two plugins are essentially the same they were many months ago. 

yes, and the problem started only after the last system upgrade in which a new version of easy (and other audio packages) was pushed - i'm not saying that easy is in fact to blame because i simply don't know, but what i do know is that disabling easy seems to fix the volume jump problems

> Either this preset has something that is not playing nice with the latest EasyEffects release or something odd happened in your PipeWire install.

certainly possible, but in the issue i linked there seems to be [another guy](https://github.com/Digitalone1/EasyEffects-Presets/issues/11#issuecomment-1318234109) on Fedora having the same problem

if you have any troubleshooting suggestions, i'm all ears (i'm not super knowledgeable with this stuff, or Linux in general)

**Comment by wwmm** [comment]
>if you have any troubleshooting suggestions, i'm all ears (i'm not super knowledgeable with this stuff, or Linux in general)

I still think that is strange to have so many plugins in this preset list. I tested it here and the sound was really weird.

**Comment by atomGit** [comment]
it's really weird for me too if the gate is enabled, but it was perfectly fine prior to the last system upgrade

i'm using the LoudnessEqualizerPE.json preset because i couldn't find any other for normalizing volume of movies that a) worked well and b) worked as an easy preset - if anyone knows of one, please share

**Comment by wwmm** [comment]
>i'm using the LoudnessEqualizerPE.json preset because i couldn't find any other for normalizing volume of movies that a) worked well and b) worked as an easy preset - if anyone knows of one, please share

In most cases when I watch movies the autogain plugin is enough for me. You may try to combine it with the `maximizer` and the `compressor` plugins. But as the autogain plugin is right now I do not feel any need at all of using other plugins to normalize movies audio.

**Comment by atomGit** [comment]
i tried the autogain plug a while back, but i'll give it another shot - thanks 

the compressor preset dev knows about this issue so maybe he can answer your other questions

**Comment by atomGit** [comment]
yes, i was using LoudnessEqualizer.json - i just linked to the wrong one when i opened the issue - i corrected the link

however... i just removed that preset, so i'm using pure easyeffects with the plugins installed with it and the same thing happens when i load the gate plugin - the audio becomes very distorted, like it's over compressed or something

**Comment by wwmm** [comment]
>the audio becomes very inconsistent - whenever there's a bit of near silence, the volume drops completely, then it takes maybe a 100ms or so until it comes back up again

That is what gates do. If the audio level is below the threshold they do will attenuate the signal. Depending on how much attenuation is being done and on the signal level you may get silence. Once the signal level is high enough again the gate will remove the attenuation.

In other words an approach based on gates will have results that will depend on the intensity of the signal getting to it. If you want to use them you will have to customize the gate threshold or the signal input level to your needs.

**Comment by Digitalone1** [comment]
@wwmm they were using the correct preset but, for unknown reasons, the values were not applied (saw that on the screenshot posted in the issue opened on my repo).

Don't know if it's a bug or whatever, but definitely not a bug with LSP Gate because if the correct values are applied, it "sounds good".

**Comment by thjderjktyrjkt** [comment]
> when easy is enabled, the volume may be different after a reboot (louder for me) - adjusting the volume, even a tiny amount, causes the level to jump back to normal and all seems ok again
>
> when using the compressor preset the problem is amplified many times - the volume can jump wildly when the volume slider is manually adjusted and there is severe distortion in the audio - this can be traced to the gate component; if the gate is disabled, all returns to normal

And i've just encountered this issue, again. just like before:

https://github.com/Digitalone1/EasyEffects-Presets/issues/11#issuecomment-1318234109

> I think i'm having the same issue, occasionally. Force stopping easyeffect then restarting again fixes the issue, until it'll occur again randomly.
> Sadly I just don't know how to reproduce, it seems to be too random. Maybe it's easyeffect's issue?

It's simply too random I can't manually reproduce it. I was listening to a music and all of a sudden the volume became very low, I touched the volume slider and it became normal again.

**Comment by Digitalone1** [comment]
This seems a general issue not related to the preset. Open a ticket on EasyEffects repo.

**Comment by wwmm** [comment]
So much has changed that is probably better to close this issue

**Comment by atomGit** [comment]
wow, yeah, sorry for not following up on this - indeed, the problem no longer exists
